src/utils/playground.ts

```diff
@@ -0,0 +1,118 @@
+import * as vscode from 'vscode';
+import micromatch from 'micromatch';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { createLogger } from '../logging';
+
+const log = createLogger('playground utils');
+
+export class FileStat implements vscode.FileStat {
+  constructor(private fsStat: fs.Stats) {}
+
+  get type(): vscode.FileType {
+    if (this.fsStat.isFile()) {
+      return vscode.FileType.File;
+    }
+    if (this.fsStat.isDirectory()) {
+      return vscode.FileType.Directory;
+    }
+    if (this.fsStat.isSymbolicLink()) {
+      return vscode.FileType.SymbolicLink;
+    }
+
+    return vscode.FileType.Unknown;
+  }
+
+  get isFile(): boolean | undefined {
+    return this.fsStat.isFile();
+  }
+
+  get isDirectory(): boolean | undefined {
+    return this.fsStat.isDirectory();
+  }
+
+  get isSymbolicLink(): boolean | undefined {
+    return this.fsStat.isSymbolicLink();
+  }
+
+  get size(): number {
+    return this.fsStat.size;
+  }
+
+  get ctime(): number {
+    return this.fsStat.ctime.getTime();
+  }
+
+  get mtime(): number {
+    return this.fsStat.mtime.getTime();
+  }
+}
+
+const getFileNames = (filePath: string): Promise<string[]> => {
+  return fs.promises.readdir(filePath);
+};
+
+const getStat = async (filePath: string): Promise<vscode.FileStat> => {
+  return new FileStat(await stat(filePath));
+};
+
+const stat = (filePath: string): Promise<fs.Stats> => {
+  return fs.promises.stat(filePath);
+};
+
+export const isPlayground = (fileUri?: vscode.Uri) => {
+  if (!fileUri) {
+    return false;
+  }
+
+  const fileNameParts = fileUri.fsPath.split('.');
```

Comment:
Small thing, fine as is, another way to do this which might be a bit simpler would be with a regex:
```typescript
return fileUri.fsPath.match(/\.(mongodb)|(mongodb\.js)$/gi);
```